develop/tutorials/articles/110-portlets/08-applying-clay-styles-to-your-app/04-clay-management-toolbar/02-filtering-and-sorting-items-with-the-management-toolbar.markdown

```diff
@@ -18,102 +19,73 @@ Creating filters and sorting results involves the following process:
-                      getPortletURL(), "orderByCol", "name");
-                    dropdownItem.setLabel(
-                      LanguageUtil.get(_request, "name"));
-                  });
-            }
-          };
+        public class MyManagementToolbarDisplayContext
+          extends BaseManagementToolbarDisplayContext {
+
+          public MyManagementToolbarDisplayContext(
+            LiferayPortletRequest liferayPortletRequest,
+            LiferayPortletResponse liferayPortletResponse,
+            HttpServletRequest request) {
+
+            super(liferayPortletRequest, liferayPortletResponse, request);
+          }
+          ...
         }
 
-4.  Add the `getOrderByCol()` Method. This method returns the column to sort by 
-    if one exist, or it sets the default column to order by if one is not found. 
-    The example below sets the default value to the `create-date` column:
+    `SearchContainerManagementToolbarDisplayContext` example:
 
-        public String getOrderByCol() {
-          if (Validator.isNotNull(_orderByCol)) {
-            return _orderByCol;
+        public class MyManagementToolbarDisplayContext
+          extends SearchContainerManagementToolbarDisplayContext {
+
+          public MyManagementToolbarDisplayContext(
+            LiferayPortletRequest liferayPortletRequest,
+            LiferayPortletResponse liferayPortletResponse,
+            HttpServletRequest request, SearchContainer searchContainer) {
+
+            super(
+              liferayPortletRequest, liferayPortletResponse, request,
+              searchContainer);
           }
+        }
 
-          _orderByCol = ParamUtil.getString(
-            _request, "orderByCol", "create-date");
+2.  Override the `getNavigationKeys()` method to return the navigation filter 
+    dropdown item(s). If your app doesn't require any navigation filters, you 
+    can just provide the *all* filter to display everything. An example 
+    configuration is shown below:
 
-          return _orderByCol;
+        @Override
+        protected String[] getNavigationKeys() {
+          return new String[] {"all", "pending", "done"};
         }
 
-5.  open the JSP view that contains the Clay Management Toolbar and set its 
-    `filterDropdownItems` attribute to the `getFilterItemsDropdownItems()` 
-    method you defined in the display context. An example configuration is shown 
-    below:
-    
-        filterDropdownItems=
-        "<%= ddmFormAdminDisplayContext.getFilterItemsDropdownItems() %>" 
+3.  override the `getOrderByKeys()` method to return the columns to order. An 
+    example configuration is shown below:
+
+        @Override
+        protected String[] getOrderByKeys() {
+          return new String[] {"name", "items", "status"};
+        }
+
+Now that you've defined the filter keys, you can provide the logic to sort the 
+columns. 
 
 ## Setting the Sorting Order for Columns [](id=setting-the-sorting-order-for-columns)
```

Comment:
All this stuff about Comparator is relevant for the management toolbar? I feel like is something related to the search container, not to the toolbar. I'd remove points 1 and 2.

Reply:
My thought was that the Comparator was needed as part of the filtering and sorting for the toolbar. Without it, the filtering and sorting wouldn't work, right?

Reply:
I'm not really sure about that. @jbalsas could you please take a look?

Reply:
Is it really needed? I know @carloslancha has applied several toolbars and I don't think he had to create one from scratch as this explains. If we often simply use an existing one, that's what should be covered here, and how to create your own comparator/sorter should be moved to some advanced sections somewhere... this looks like it should affect SearchContainer and not ManagementToolbar... 🤔

Reply:
Ok. Will do. Thanks @jbalsas @carloslancha